Sprint-2/stretch/count-words.js

```diff
@@ -26,3 +26,25 @@
 
 3. Order the results to find out which word is the most common in the input
 */
+
+function countWords(str) {
+  str = str.toLowerCase();
+
+  const words = str.split(' ');
+
+  const wordCounts = {};
+
+  for (const word of words) {
+    if (wordCounts[word]) {
+      wordCounts[word] += 1;
+    } else {
+      wordCounts[word] = 1;
+    }
+  }
+   const sortedWordCounts = Object.entries(wordCounts).sort((x, y) => x[1] - y[1]);
+   const sortedObject = Object.fromEntries(sortedWordCounts);
+
+  return wordCounts;
+}
```

Comment:
Good work, were there any edge questions you were thinking of when doing this problem solving?

Reply:
I’ve updated the code to handle empty strings and strings with extra spaces. Now, the function returns an empty result for empty strings and treats multiple spaces as one and counts the words correctly. There are also other edge cases like Case Sensitivity, Punctuation, Single Word String...